tribe_client/utils.py

```diff
@@ -395,17 +395,61 @@ def obtain_token_using_credentials(username, password, client_id,
     return tribe_response['access_token']
 
 
+def download_organism_public_genesets(organism, creator_username=None):
+    """
+    Function to download all the public genesets available for a given
+    organism, and optionally, for a given creator username.
+
+    Arguments:
+    organism -- A string, of the scientific name for the desired species.
+
+    creator_username --  Optional argument, a string. Filter the genesets
+    downloaded from Tribe to get only the ones created by the user with
+    this username.
+
+    Returns:
+    all_public_genesets -- A dictionary of GO, KEGG and DO terms available
+    in Tribe for the specified organism (and optionally, creator_username).
+    """
+
+    # *Note: Tribe does not like requests for more than 1500
+    # genesets at a time, so use this as the 'limit' parameter.
+    request_params = {
+        'show_tip': 'true', 'limit': '1500',
+        'organism__scientific_name': organism
+    }
+
+    if creator_username:
+        request_params['creator__username'] = creator_username
+
+    request_params['title__startswith'] = 'GO'
+    go_public_genes = retrieve_public_genesets(request_params,
+                                               retrieve_all=True)
+
+    request_params['title__startswith'] = 'KEGG'
+    kegg_public_genes = retrieve_public_genesets(request_params,
+                                                 retrieve_all=True)
+
+    request_params['title__startswith'] = 'DO'
+    omim_public_genes = retrieve_public_genesets(request_params,
+                                                 retrieve_all=True)
```

Comment:
If I understand correctly, you could write a loop like this:
```
pairs = {'GO': 'Gene Ontology', 'KEGG': 'KEGG', 'DO': 'OMIM'}
all_public_genesets = dict()
for k, v in pairs.items():
    request_params['title__startswith'] = k
    all_public_genesets[v] = retrieve_public_genesets(request_params, retrieve_all=True)
```
But since there are only three iterations in the loop, it is up to you.

Reply:
Ah, it's true - actually that does seem like a more "pythonic" way to do it.

Reply:
Good idea, @dongbohu, since a lot of the code does repeat itself!